joomla/joomla-cms
# Allow jForm to support data attribute

Pull Request for Issue #16034 

### Summary of Changes
- [HTML5](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5) is designed with extensibility in mind for data that should be associated with a particular element but need not have any defined meaning. data-* attributes allow us to store extra information on the standard, semantic HTML elements without other hacks such as non-standard attributes, extra properties on DOM.

- However, custom attributes have no special meaning generally and are only special to the owner's application. They can be used to simplify an application's logic. By allowing JForm to support data attributes, it can be greatly aid development of extensions both core and third party.

### Testing Instructions
-  Add data attribute in XML form, for example
     -  Open administrator/components/com_users/models/forms/user.xml file
     -   Add data attribute in xml field, like
          <pre><code> &lt;field name="name" 
                 type="text"
	             description="COM_USERS_USER_FIELD_NAME_DESC"
	             label="COM_USERS_USER_FIELD_NAME_LABEL"
         	required="true"
                 size="30"
                 data-action-type="click"  // add data attribute like this
            /&gt;
           </code ></pre>

- Login into administrator
- Click on menu Users => Manage => Add New User
- Inspect the filed = *Name* and check input type html

### Expected result
  * Output should be like this 
      <pre><code> &lt;input type="text" 
            name="jform[name]" 
            id="jform_name" 
            value="" 
            class="required invalid" 
            size="30" 
            required="required" 
            aria-required="true" 
            data-action-type="click"  // output 
            autocomplete="off" 
            aria-invalid="true"
       /&gt;
         </code ></pre>
   * Output snapshot
![screenshot from 2017-05-15 19 27 38](https://cloud.githubusercontent.com/assets/16605829/26061183/cf507526-39a4-11e7-8aa9-a7a562a006db.png)

### Actual result

* Added data attribute is missing in final output
      <pre><code> &lt;input type="text" 
                      name="jform[name]" 
                      id="jform_name" 
                      value="" 
                      class="required invalid" 
                      size="30" 
                      required="required" 
                     aria-required="true" 
                     autocomplete="off" 
                     aria-invalid="true"
       /&gt;
         </code ></pre>


### Documentation Changes Required
